components/dashboard/src/prebuilds/detail/PrebuildTaskTab.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Prebuild, PrebuildPhase_Phase } from "@gitpod/public-api/lib/gitpod/v1/prebuild_pb";
+import { Suspense, useEffect } from "react";
+import { usePrebuildLogsEmitter } from "../../data/prebuilds/prebuild-logs-emitter";
+import React from "react";
+import { useToast } from "../../components/toasts/Toasts";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { TabsContent } from "@podkit/tabs/Tabs";
+import { PersistedToastID } from "./PrebuildDetailPage";
+
+const WorkspaceLogs = React.lazy(() => import("../../components/WorkspaceLogs"));
+
+type Props = {
+    taskId: string;
+    prebuild: Prebuild;
+    onLogNotFound: () => void;
+};
+export const PrebuildTaskTab = ({ taskId, prebuild, onLogNotFound }: Props) => {
+    const { emitter: logEmitter, disposable: disposeStreamingLogs } = usePrebuildLogsEmitter(prebuild.id, taskId);
+    const { toast } = useToast();
+
+    useEffect(() => {
```

Comment:
This useEffect looks fishy: Why do we need it? `usePrebuildLogsEmitter` should be able to cleanup after itself on unmount. `disposeStreamingLogs` should only be needed for on-demand cancallation (like in the effect below, on "aborted")

Reply:
You're totally right. This here should not care about disposal